frontend/views/containers/chatroom/MessagePoll.vue

```diff
@@ -82,6 +104,64 @@ export default ({
     },
     addEmoticon (emoticon) {
       this.$emit('add-emoticon', emoticon)
+    },
+    switchOnChangeMode () {
+      this.ephemeral.isChangeMode = true
+    },
+    switchOffChangeMode () {
+      this.ephemeral.isChangeMode = false
+    },
+    checkAndSetupPollExpirationTimer () {
+      const markPollClosed = () => {
+        try {
+          sbp('gi.actions/chatroom/closePoll', {
+            contractID: this.currentChatRoomId,
+            data: { hash: this.messageHash }
+          })
```

Comment:
From my knowledge, it's not necessary for `try ... catch` to have `await` inside it, to catch any error. (right?) if there is any error or exception thrown during the `sbp` operation that leads to the failure of closing the poll, it will let the user know by showing the banner.

Reply:
@SebinSong It is necessary:

```
$ node
> (async () => { try { (async () => { throw 'asdf'; })() } catch (e) { console.log('caught:', e) } })()
Promise {
  undefined,
  [Symbol(async_id_symbol)]: 946,
  [Symbol(trigger_async_id_symbol)]: 5
}
> Uncaught 'asdf'
```

The try/catch is just a rewritten syntax sugar over `new Promise(() => { .. }).catch(..)`, and if you don't put the `await` you don't properly chain the `.catch()`

Reply:
@taoeffect 
```
const throwError = () => {
  throw new Error('random error!')
}

const anotherFunction = () => {
  try {
    throwError()
  } catch (error) {
    console.error('error thrown: ', error);
  }
}

anotherFunction()
```

Thought it still contradicted my experience of using it, so I just wrote/run this test script and it does seem to catch exceptions in synchronous operation too.

Reply:
@SebinSong Your example does not use a Promise. The reason we call `await` here is because a promise is involved.

Reply:
@taoeffect got it. will attach the `await` keyword then.